### PR TITLE
Publish to npm/GHPR

### PR DESCRIPTION
**Pull request by jasonkarns**
Adds workflow to publish as an npm package to the npmjs registry and github package registry.
The workflow has been tested and works (pre-release alphas have been published to both registries).
This also incorporates a host of administrative changes regarding orgs and permissions on npm.

Remaining todo: determine what the release workflow will actually be and change the trigger on this workflow accordingly. My assumption is that we'll want the creation of a Release on github to be manual (while we still manually create release notes, etc). That is to say, we probably _don't_ want the release process to be wholly automated off of, say, a git tag event. (yet)

- [x] I have reviewed the [Contributor Guidelines][contributor].
- [x] I have reviewed the [Code of Conduct][coc] and agree to abide by it

[contributor]: https://github.com/bats-core/bats-core/blob/master/docs/CONTRIBUTING.md
[coc]:         https://github.com/bats-core/bats-core/blob/master/docs/CODE_OF_CONDUCT.md

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+  github-npm:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          registry-url: "https://npm.pkg.github.com"
+      - name: scope package name as required by GHPR
+        run: npm init -y --scope ${{ github.repository_owner }}
```
Comment (martin-schulze-vireso) [inline .github/workflows/release.yml:27]: I have no npm experience, why is this necessary here, but not in the other job?

**Review by martin-schulze-vireso — APPROVED** [review]
I am not familiar with npm but the GHA workflow looks good to me.

File: .github/workflows/release.yml (hunk 1 above)
Comment (jasonkarns) [reply .github/workflows/release.yml:27]: On npm, the package is published unscoped (as `bats`). The GHPR, on the other hand, doesn't support or allow unscoped packages. The package name, therefore, must be scoped to the organization: `@bats-core/bats`.

This was the simplest way I could think of to change the package.json such that the package name is scoped prior to publishing to GHPR.

An alternative would be to deprecate the unscoped `bats` package and publish it scoped on both registries. That might be something we should consider.

Comment (martin-schulze-vireso) [reply .github/workflows/release.yml:27]: I'd say we can defer this discussion to get v1.3.0 out the door. Maybe you want to open an issue for that?
